lib/internal/blob.js

```diff
@@ -123,7 +123,7 @@ function getSource(source, endings) {
   // BackingStores are going to be detached and owned by
   // the Blob. We also don't want to have to worry about
   // byte offsets.
-  source = new Uint8Array(source);
+  source = new Uint8Array(source.buffer.slice(0), source.byteOffset);
```

Comment:
Actually this is better, as it would avoid copying unnecessary bytes.

```suggestion
  source = new Uint8Array(source.buffer.slice(source.byteOffset));
```